ghpghx/resources.py

```diff
@@ -0,0 +1,207 @@
+            if ghpghxM.heating_thermal_load_mmbtu_per_hr in [None, []]:
+                ghpghxM.heating_thermal_load_mmbtu_per_hr = list(np.array(ghpghxM.heating_fuel_load_mmbtu_per_hr) * \
+                                                ghpghxM.existing_boiler_efficiency)
+            # Call PVWatts for hourly dry-bulb outdoor air temperature
+            if ghpghxM.ambient_temperature_f in [None, []]:
+                try:
+                    pvwatts_inst = PVWatts(tilt=45, latitude=ghpghxM.latitude, longitude=ghpghxM.longitude)
+                    amb_temp_c = pvwatts_inst.response["outputs"]["tamb"]
+                    ghpghxM.ambient_temperature_f = list(np.array(amb_temp_c) * 1.8 + 32.0)
+                except Exception:
+                    return400(data, {"Error": "Error in PVWatts call"})
+        except ImmediateHttpResponse as e:
+            raise e
+        except Exception:
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            data["status"] = ('Internal Server Error during input validation. No GHPGHX task has been created. '
+                              'Please check your POST for bad values.')
+            data['inputs'] = bundle.data
+            data['messages'] = {}
+            data['messages']['error'] = "Unexpected Error."
+            log.error("Internal Server error: " + data['messages']['error'])
+            raise ImmediateHttpResponse(HttpResponse(json.dumps(data),
+                                                     content_type='application/json',
+                                                     status=500))  # internal server error
+
+        try:
+            # Save the instance of the GHPGHXModel to the database
+            ghpghxM.save()
+        except Exception:
+            log.error("Could not create and save ghp_uuid: {}\n Data: {}".format(ghp_uuid, data))
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            data["status"] = "Internal Server Error during saving of inputs. Please see messages."
+            data['messages'] = "Unexpected Error."
+            log.error("Internal Server error: " + data['messages'])
+            raise ImmediateHttpResponse(HttpResponse(json.dumps(data),
+                                                     content_type='application/json',
+                                                     status=500))  # internal server error
+        
+        data["inputs"] = ModelManager.make_response(ghp_uuid=ghp_uuid)["inputs"]
+        # Remove extra inputs used above but not expected in ghpghx_inputs.jl
+        data["inputs"].pop("latitude", None)
+        data["inputs"].pop("longitude", None)
+        data["inputs"].pop("heating_fuel_load_mmbtu_per_hr", None)
+        data["inputs"].pop("existing_boiler_efficiency", None)
+        data["status"] = 'Solving for GHX Size...'
+
+        try:
+            # TODO Make this a celery task so we don't have to wait before getting ghp_uuid back in response
```

Comment:
@Bill-Becker do you still plan on making the ghpghx call a celery task? (If not it looks like the run_ghpghx_sizing.py and process_ghpghx_results.py files are obsolete.)

Reply:
These files have been removed.

Reply:
No, I removed this comment.